.github/workflows/publish-to-test-pypi.yml

```
@@ -0,0 +1,118 @@
+name: Build sdist and wheel and publish to PyPI and TestPyPI
+
+#on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build_sdist:
+    name: Build sdist
+    runs-on: ubuntu-latest
+    steps:
+
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: Install ninja
+      run: pipx install ninja
+
+    - name: Build sdist
+      run: pipx run build --sdist
+
+    - name: Check metadata
+      run: pipx run twine check --strict dist/*
+
+    - uses: actions/upload-artifact@v3
+      with:
+        path: dist/*.tar.gz
+
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
```

```suggestion
    strategy:
      fail-fast: false
```

To not cancel jobs, other jobs on the first fail. In final version could be removed but simplify debugging.